Adyen/UI/Form/Items/Value Pickers/Abstract/BaseFormValuePickerItemView.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+
+/// Represents a picker item view.
+open class BaseFormValuePickerItemView<T: CustomStringConvertible & Equatable>: FormValueItemView<BasePickerElement<T>,
+    FormTextItemStyle,
+    BaseFormValuePickerItem<T>>,
+    UIPickerViewDelegate,
+    UIPickerViewDataSource {
+
+    /// The underlying `UIPickerView`.
+    internal lazy var pickerView: UIPickerView = {
+        let pickerView = UIPickerView()
+        pickerView.delegate = self
+        pickerView.dataSource = self
+        pickerView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        return pickerView
+    }()
+
+    /// Initializes the picker item view.
+    ///
+    /// - Parameter item: The item represented by the view.
+    public required init(item: BaseFormValuePickerItem<T>) {
+        super.init(item: item)
+        initialize()
+        didSelection(selectionValue: item.value.wrappedValue)
+    }
+
+    /// :nodoc:
+    override open var canBecomeFirstResponder: Bool { true }
+
+    /// :nodoc:
+    override open func becomeFirstResponder() -> Bool {
+        inputControl.becomeFirstResponder()
+    }
+
+    /// :nodoc:
+    override open func resignFirstResponder() -> Bool {
+        inputControl.resignFirstResponder()
+    }
+
+    // MARK: - Abstract
+
+    internal func getInputControl() -> PickerTextInputControl {
+        BasePickerInputControl(inputView: pickerView, style: item.style.text)
+    }
+
+    internal func updateSelection() {
+        inputControl.label = item.value.wrappedValue.description
+    }
+
+    internal func initialize() {
+        addSubview(inputControl)
+        inputControl.translatesAutoresizingMaskIntoConstraints = false
+        inputControl.preservesSuperviewLayoutMargins = true
+        (inputControl as UIView).adyen.anchor(inside: self)
```

Comment:
can we do it without casting here?

Reply:
No unfortunately - inputControl is generic and Swift complains:
```Member 'adyen' cannot be used on value of protocol type 'PickerTextInputControl'; use a generic constraint instead```

This is a soft casting, tho